lib/access_token_agent/token.rb

```diff
@@ -21,7 +21,7 @@ def valid?
     private
 
     def validate_response(auth_response)
-      unless auth_response['token_type'] == 'bearer'
+      unless auth_response['token_type'].downcase == 'bearer'
```

Comment:
Do we have a `nil` check somewhere? Otherwise this code could fail if `auth_response['token_type']` was nil.

Reply:
Added a nil check and a related spec.